pages/index.tsx

```diff
@@ -2,25 +2,81 @@ import type { NextPage } from "next";
 import React, { useState, useEffect } from "react";
 import { EuiText, EuiFieldText, EuiFlexGroup, EuiFlexItem, EuiButton } from "@elastic/eui";
 import { useRouter } from 'next/router';
+import Link from 'next/link'
+import { database } from '../firebaseConfig';
+import { setDoc, doc, collection, getDocs, where, query, DocumentData } from "firebase/firestore";
+
+const roomsCollection = collection(database,'rooms');
 
 const Home: NextPage = () => {
   const router = useRouter();
-  const [newRoom, setNewRoom] = useState("");
+  const [newRoom, setNewRoom] = useState('');
   const [joinRoom, setJoinRoom] = useState('');
+  const [roomList, setRoomList] = useState<any[]>([]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const getActiveRooms = () => {
+    let docs: React.SetStateAction<any[]> = [];
+    getDocs(roomsCollection)
+      .then((data) => {
+        data.forEach((doc) => {
+          let id = doc.id;
+          let rd = doc.data();
+          rd.id = id;
+          docs.push(rd);
+        })
+        setRoomList(docs)
+      })
+  };
```

Comment:
This function queries FB and returns an array of active rooms. It then sets the `roomList` state to match FB.